### PR TITLE
[REVIEW] Workflow refactor

### PR DESCRIPTION
**Pull request by benfred**
In preparation for merging the dask/cudf workflow objects this is a first step
at simplifying some workflow implementation details:

* Removes the OperatorRegistry class, and the 'ops_args' member on the workflow
* Moves the master_task_list and task_sets from being member variables of the workflow class to being local variables of the load_config method
* Removes the operator registration code and stat_ops/df_ops/feat_ops  members
* Removes the 'to_cpu' parameter on the workflow class

### CODE REVIEW

**Comment by benfred** [comment]
rerun tests

File: nvtabular/workflow.py [hunk 1]
```diff
@@ -65,17 +65,7 @@ class BaseWorkflow:
...
-        New feature engineering operators can be added
-        later using the other functions
-    stat_ops : list of stats operator objects
-        Feature engineering ops objects such as
-        Categorify and Normalize. New feature engineering
-        operators can be added later using the other functions.
-    df_ops : list of data frame operator objects
-    to_cpu : bool, default True
-    config : bool
+    config : object
```
Comment (rjzamora) [inline nvtabular/workflow.py:68]: Maybe we can explain the `config` argument a bit more here?

File: nvtabular/workflow.py [hunk 2]
```diff
@@ -85,21 +75,13 @@ def __init__(
         cat_names=None,
         cont_names=None,
         label_name=None,
-        feat_ops=None,
-        stat_ops=None,
-        df_ops=None,
-        to_cpu=True,
         config=None,
+        # TODO: move these out
```
Comment (rjzamora) [inline nvtabular/workflow.py:79]: Just to clarify - The plan is to turn "exporting" into a separate Op in the future?

File: nvtabular/workflow.py (hunk 1 above)
Comment (benfred) [reply nvtabular/workflow.py:69]: I think we should actually make config a class in its own right (like Alec suggested here https://github.com/NVIDIA/NVTabular/issues/15).

File: nvtabular/workflow.py (hunk 2 above)
Comment (benfred) [reply nvtabular/workflow.py:80]: Yes, we're planning on having separate 'export' ops in the future. The default 'export_path' does get used in the 'phases_export' function right now - though that could probably be removed too

File: nvtabular/workflow.py [hunk 3]
```diff
@@ -641,23 +556,16 @@ def run_ops_for_phase(self, gdf, tasks, record_stats=True):
...
-                )
-            elif op._id in self.df_ops:
-                gdf = self.df_ops[op._id].apply_op(
-                    gdf,
-                    self.columns_ctx,
-                    cols_grp,
-                    target_cols=target_cols,
-                    stats_context=self.stats,
-                )
+
```
[the remark this answers is not in the document]
Comment (benfred) [reply nvtabular/workflow.py:532]: I think that this is ok now - the tests + examples seem to all work with these changes.

**Comment by benfred** [comment]
> It's not related to any change you made, but one thing I noticed while looking this over is that the `config` argument seems to be a missed opportunity. It might be nice to consolidate the `load_stats` and `load_config` functionality so that the user can pass in a single file (as `config`) to specify _both_ the workflow configuration and known statistics. An obvious application would be "online-only" preprocessing.

Lets examine this in a future PR - I agree that we probably want to make online-only / inference support seamless, but have some concerns about serializing the ops that are part of the current config object.

**Pull request merged**